### ISSUE TITLE
NestedMenuItem onClick event not behaving as expected

### ISSUE THREAD

**Issue by choieunsoh** [steviebaa/mui-nested-menu#7]
Hi,

I work with hierarchy submenu component look like
```js
const NestedCategoryItem = ({ id, label, children = [], depth = 0 }: NestedCategoryItemProps) => {
  const { setValue } = useCategoryFilterContext();

  return (
    <NestedMenuItem
      label={label}
      parentMenuOpen={true}
      rightIcon={children.length > 0 ? <ChevronRightIcon /> : null}
      style={{ marginRight: 8 }}
      onClick={() => setValue({ id, label })} // <<< this line
    >
      {!!children.length &&
        children.map((child) => <NestedCategoryItem key={child.id} {...child} depth={depth + 1} />)}
    </NestedMenuItem>
  );
};
```

Then I click on deeper submenu and there are no options to expose the submenu that I just selected.

Would be possible to expose the event like this?

Thanks.

**Comment by steviebaa** [comment]
Hi @choieunsoh,

If I understand correctly, you are trying to access the event of the MUI MenuItem used within the `NestedMenuItem` component.

If you have a look at the source file here, you can see that the structure of the component is essentially just an `IconMenuItem` wrapped in a div container. I think the issue you're having here is that, the click event is captured by the div container.

You can add props to the NestedMenuItem via the prop `ContainerProps` and this will apply them to that container. So something like the code below will trigger the event when clicked. The only issue you might have here is that the children of that container will also trigger that event.

Are you able to work with this?

```
export const NestedCategoryItem = ({children}: {children: React.ReactNode}) => {
  const clickHandler = (e) => {
    console.log('clicked');
    console.log(e);
  };

  return (
    <NestedMenuItem
      label="Test Category"
      parentMenuOpen={true}
      rightIcon={React.Children.count(children) > 0 ? <ArrowRightIcon /> : null}
      style={{marginRight: 8}}
      ContainerProps={{
        onClick: clickHandler, // <<< this line
      }}
    >
      {React.Children.count(children) &&
        React.Children.map(children, child => child)}
    </NestedMenuItem>
  );
};
```

**Comment by choieunsoh** [comment]
Hi @steviebaa ,

Yes, I faced an issue that you mentioned before I posted.

I have 3 levels submenu and if I click a leaf submenu, it causes fire 3 events (3-level click > 2-level click > 1-level click)
Then the value that I received is the 1-level submenu value.

Thanks.

**Comment by steviebaa** [comment]
@choieunsoh, we can probably just remove [this line](https://github.com/steviebaa/mui-nested-menu/blob/main/src/mui-nested-menu/components/IconMenuItem.tsx#L42) from `IconMenuItem`. The click would be getting set from the `MenuItemProps` spread above and then overwritten with undefined. 

Even if we get rid of that, then it shouldn't be a breaking change because the `...props` spread on line 43. 

Would you like to open a PR for it?

**Comment by choieunsoh** [comment]
Hi @steviebaa ,

I try to remove 'onClick' that you mentioned above then I faced issue like below
There is no `'MenuItemProps.onClick'`
![image](https://user-images.githubusercontent.com/5287915/171127204-9e13bada-7cbd-462a-be1a-a320996760a4.png)

**Comment by steviebaa** [comment]
Published as [2.0.3](https://www.npmjs.com/package/mui-nested-menu/v/2.0.3)
